File: kody_2023/2019_2A/potega.cpp
/*
 * silnia.cpp
 */


#include <iostream>
using namespace std;

// x0 = 1 dla x=/0
// x1 = x
// xn = x * ... * x(n-czynników)

float potega_it(float x, int n) {
    float potega = 1;
    for (int i=0; i < n; i++) {
        potega = potega * x;
    }
    return potega;
}

// x0 = 1 dla x=/0
// xn = x(n-1) * x, n > 0

float potega_re(float x, int n) {
    if (n == 0)
        return 1;
    else
        return potega_re(x, n-1) * x;
}

//~potega_re(2, 4);
//~potega_re(2, 3) * 2;
//~potega_re(2, 2) * 2;
//~potega_re(2, 1) * 2;
//~potega_re(2, 0) = 1;
//~1 * 2 = 2;
//~2 * 2 = 4;
//~4 * 2 = 8;
//~8 * 2 = 16;

int main(int argc, char **argv)
{
	float x;  // podstawa
    int n;  //wykładnik
    cout << "Podaj podstawę: " << endl;
    cin >> x;
    cout << "Podaj wykładnik: " << endl;
    cin >> n;
    cout << "Potęga: " << potega_it(x, n) << endl;
    cout << "Potęga: " << potega_re(x, n) << endl;
	return 0;
}

File: kody_2025/3A_inf_r/algorytmy/potega.py
# 2^0 = 1
# 2^4 = 2 * 2 * 2 * 2

def potega_it(a, n):
    wynik = 1
    for i in range(n):
       wynik = wynik * a
    return wynik


def potega_re(a, n):
    print(a, n)
    if n == 0:
        return 1
    return potega_re(a, n-1) * a

def main():

    # pobierz podstawę a
    a = int(input('Podaj podstawę: '))
    # pobierz wykładnik n
    n = int(input('Podaj wykładnik: '))

    print(potega_it(a, n))
    print(potega_re(a, n))


main()
